test/unit/commands/plugin-install.test.js

```diff
@@ -59,4 +59,43 @@ describe('test/unit/commands/plugin-install.test.js', async () => {
     });
     expect(serverlessFileObj.plugins).to.include(pluginName);
   });
+
+  it('should not add plugin to serverless file if it is already present in configuration but configured behind a variable', async () => {
+    const installPlugin = proxyquire('../../../commands/plugin-install', {
+      'child-process-ext/spawn': sinon.fake(),
+    });
+
+    const fixture = await fixturesEngine.setup('function', {
+      configExt: {
+        plugins: ['${self:custom.pluginName}'],
+        custom: {
+          pluginName,
+        },
+      },
+    });
+
+    const configuration = fixture.serviceConfig;
+
+    // Simulate that the variable has been resolved
+    configuration.plugins = [pluginName];
+    const sDir = fixture.servicePath;
```

Comment:
Let's use meaningful names, so `serviceDir` instead of `sDir` (when reading the code later, it's hard to assume  what `sDir` is about)

Reply:
I've shortened the name here on purpose because otherwise it was shadowing `serviceDir` from upper scope - I'll think of something better